index.js

```diff
@@ -27,23 +28,27 @@ function parseTorrent (torrentId) {
     const torrentObj = magnet(torrentId)
 
     // infoHash won't be defined if a non-bittorrent magnet is passed
-    if (!torrentObj.infoHash) {
+    if (!torrentObj.infoHash && !torrentObj.infoHashV2) {
       throw new Error('Invalid torrent identifier')
     }
 
     return torrentObj
   } else if (typeof torrentId === 'string' && (/^[a-f0-9]{40}$/i.test(torrentId) || /^[a-z2-7]{32}$/i.test(torrentId))) {
-    // if info hash (hex/base-32 string)
+    // if v1 info hash (hex/base-32 string)
     return magnet(`magnet:?xt=urn:btih:${torrentId}`)
+  } else if (typeof torrentId === 'string' && /^1220(.{64})/.test(torrentId)) {
+    // if v2 info hash (base-32 string)
+    return magnet(`magnet:?xt=urn:btmh:${torrentId}`)
   } else if (Buffer.isBuffer(torrentId) && torrentId.length === 20) {
-    // if info hash (buffer)
+    // if v1 info hash (buffer)
     return magnet(`magnet:?xt=urn:btih:${torrentId.toString('hex')}`)
   } else if (Buffer.isBuffer(torrentId)) {
     // if .torrent file (buffer)
     return decodeTorrentFile(torrentId) // might throw
-  } else if (torrentId && torrentId.infoHash) {
+  } else if (torrentId && (torrentId.infoHash || torrentId.infoHashV2)) {
     // if parsed torrent (from `parse-torrent` or `magnet-uri`)
-    torrentId.infoHash = torrentId.infoHash.toLowerCase()
+    if (torrentId.infoHash) torrentId.infoHash = torrentId.infoHash.toLowerCase()
+    if (torrentId.infoHashV2) torrentId.infoHashV2 = torrentId.infoHashV2.toLowerCase()
```

Comment:
Is it even possible to reach this point with a capitalized hex hash?

Reply:
It might, magnet does not lower case it